### ISSUE TITLE
In Windows "start projectname.komodoproject" doesn't open "projectname"

### ISSUE THREAD

**Issue by hobbsd** [Komodo/KomodoEdit#1683]
### Short Summary
In Windows 10 commandline (cmd) entering `start projectname.komodoproject` doesn't open the "projectname" project.

### Steps to Reproduce
Open cmd.exe window and go to project directory where there's an appropriate "projectname.komodoproject" file; type `start projectname.komodoproject`.

### Expected results
KE starts with "projectname" project open.

### Actual results
KE started with previously opened project plus one other window with no project open in it.

### Platform Information
Komodo Edit, version 10.0.0, build 17209, platform win32-x86.
Built on Thu May 12 00:43:26 2016.

### Additional Information
The previous KE session had two projects open in two windows.

**Comment by Naatan** [comment]
What is this "start" executable? I'm not familiar with it.

**Comment by mitchell-as** [comment]
It's sort of like double-clicking on the file in question. When you "start" a URL, it opens in the default browser for example.

@hobbsd I'm not too familiar with the intricacies of "start" myself, but what happens if you do `start "" projectname.komodoproject`?

**Comment by cgchoffman** [comment]
This appears to be working fine for me.
`start myProject.komodoproject` started Komodo (one window) and opened "myProject" in the window.  I ran it again with the window open, with another project and that project was opened in Komodo.

@hobbsd, is this reproducible?  I suspect komodo trying to restore a previous, multi window workspace.  I know I came a across a bug (and have a fix pending) where multi-window restore fails due to a race while restoring projects in multiple windows.

**Comment by hobbsd** [comment]
It is reproducible, and you're right that it's related to multi window workspace. Open Project1 and Project2 in KE, then close KE using `alt-f4`. In the cmd window run `start Project3.komodoproject`. Two KE windows will open, one having either Project1 or Project2 open, the other having no project open but the root dir of Project1 or Project2 (the one that didn't open in the other window) open in Places.

**Comment by cgchoffman** [comment]
And neither of those is `myProject.komodoproject`? Or is one of them the arg used in commandline?

**Comment by hobbsd** [comment]
Neither is the one from the commandline.

**Comment by cgchoffman** [comment]
Are you up for trying to patch your install with a fix I have?  You can't just run a patch tool to apply it.  You'll need to unzip the komodo.jar in your install, apply the patch, then rezip.

**Comment by hobbsd** [comment]
I'm willing to give it a shot.

**Comment by hobbsd** [comment]
If possible it might make the most sense to not restore multi-windows/projects workspaces when a project file is used to start KE.

**Comment by cgchoffman** [comment]
I was just about to reply with the same thought.  Our restore code should handle commandline arguments properly.  Do you ( @hobbsd ) mind altering the title of the bug to reflect that? ie. "Komodo should not restore workspace when process started with file args" or something like that.  @Naatan, does that makes sense?

**Comment by Naatan** [comment]
@cgchoffman I think in this case we should prompt them to restore the last workspace or not. Because otherwise it will be lost entirely, which might not be desirable either.

**Comment by cgchoffman** [comment]
That makes sense.  Thanks for moving the milestone btw.

**Comment by hobbsd** [comment]
Maybe instead there should be a prompt about replacing the stored workspace at exit, seeing as opening KE via a project file is fairly explicit in its intent.

**Comment by Naatan** [comment]
I don't think on exit would make any sense. You don't know on exit what the context will be of your starting Komodo the next time.

**Comment by hobbsd** [comment]
Personally, I'd rather not be asked at start about restoring the last workspace when it's obvious that I want to work on the project I'm opening.

**Comment by Naatan** [comment]
You can turn off workspace restoring entirely if you don't like it, this would only be for people who use this mechanic.

**Comment by hobbsd** [comment]
It's not that I don't like it -- I use it currently -- I'd just prefer it to work differently, I guess. The way that makes sense to me is that if I open a project file, KE would just open that project; and if I open the KE application, it would restore the workspace. Just one KomodoEdit user's opinion, of course. :)

**Comment by Naatan** [comment]
I think you are using KomodoEdit too much like a quick launch app that you close and open 50 times a day. The intended UX for Komodo is not that you open it quickly to edit a file and then close it entirely. While you can certainly use it this way, this is not what Komodo is designed for.

**Comment by cgchoffman** [comment]
I agree with @hobbsd.  If I'm explicitly starting Komodo with a Komodo project file then it's pretty clear what I want to happen; I want to start working on that project.  

If someone is that attached to a workspace then they should explicitly save it using custom workspaces.

**Comment by hobbsd** [comment]
@Naatan I tend to open KE once per day, if not less (i.e., I leave it open for several days) -- I have lighter weight editors for quick open & close edits. Sometimes I do need to restart KE if the codeintel gets weird, and then it's nice to have the workspace restored.

**Comment by hobbsd** [comment]
One more scenario for this issue, in case it's helpful: yesterday I closed KE with a workspace of two windows, Project1 and Project2. Today I started KE with `start Project1`, and KE opened the two saved workspace windows, one of which had Project1 and the other had a dialog saying it couldn't open Project1 because it was already open in another window.

**Comment by Naatan** [comment]
We just pushed a fix for this issue. Note please create new issues for new problems, even if they seem somewhat related.

**Comment by tonyattwood** [comment]
Hi,

I just wanted to share with you some pics of my favourite place, it;s so awesome!  Here is the  link for  the pics <http://fact.gga100.com/e4dfzn>

Looking forward, tony.attwood@ntlworld.com
